### ISSUE TITLE
Let there be fade in/out with custom duration for audio and video tracks 

### ISSUE THREAD

**Issue by raindropsfromsky** [OpenShot/openshot-qt#5436]
Currently, the fade in and fade out functions for audio and video have only two settings: long and short.
This is too restrictive.

Often the user wants a longer duration for this setting (often by trial and error).
Please add "fade in" and "fade out" options with custom duration 
(an input box should pop up and allow the user to enter a value). 

(I have also made another suggestion to let the user drag the keyframe for such changes. That will complement this feature.)

**Comment by Colorjet3** [comment]
Hello @raindropsfromsky.  

1.  I am assuming your are talking about the option where you right click on a clip, select Fade, then select one of the options and then Fade in (**Fast**) or Fade In (**Slow**), etc.

Correct?

2.  In the Transitions tab there is a Fade transition.  You can drag this onto a clip.  By default the duration is 10 seconds.

You can shorten or lengthen this Fade either by the mouse or looking at the Properties and change the End attribute.

You can also shift the transition if the clip you dragged on is longer than 10 seconds.  To achieve Fade Out instead of Fade In you would right click on the Fade transition and select "Reverse Transition".

I am thinking that the Fade from the Transition is already customizable so is this good enough.

What are your thoughts?  Do we need both options then, and if so Why?

Thanks in advance!

**Comment by raindropsfromsky** [comment]
Yes, I was referring to the quick context menu with 'fast' and 'slow' options.

Actually, what I love about Openshot is the context menu that gives the most needed options right there.
It's really a very user-friendly design.

I don't mind if the same thing can be done with a Fade transition.
(That's just like how Openshot allows us to add a media file in multiple ways.)

That's why I thought that a custom value option should also be there.

But if my other suggestion to let the user drag any keyframe marker is implemented, then this suggestion can be dropped altogether!

**Comment by Colorjet3** [comment]
Thank you @raindropsfromsky for the feedback.

**Comment by ekw** [comment]
>     1. I am assuming your are talking about the option where you right click on a clip, select Fade, then select one of the options and then Fade in (**Fast**) or Fade In (**Slow**), etc.
> 
> Correct?
> 
>     2. In the Transitions tab there is a Fade transition.  You can drag this onto a clip.  By default the duration is 10 seconds.
> 
> You can shorten or lengthen this Fade either by the mouse or looking at the Properties and change the End attribute.
> 
> You can also shift the transition if the clip you dragged on is longer than 10 seconds. To achieve Fade Out instead of Fade In you would right click on the Fade transition and select "Reverse Transition".
> 
> I am thinking that the Fade from the Transition is already customizable so is this good enough.
> 
> What are your thoughts? Do we need both options then, and if so Why?
> 
> Thanks in advance!

I'm not the OP, but I prefer using Fade In/Out because the fade effect stays with the clip when it is moved.  I just would also like to see the fade duration be customizable/editable.  

I do not like using the Fade transition from the Transitions tab because it creates a separate object that must be separately managed (i.e., when the clip is moved, the fade transition object must also be moved).  It's extra work that I feel should be unnecessary.

**Comment by ekw** [comment]
@raindropsfromsky 
I added preferences to customize the lengths for fade fast/slow:  
https://github.com/ekw/openshot-qt/commit/1214e7e0e9814b2f081baa031275ad01943a09fb

Setting these new preferences doesn't affect existing fades, only future ones, so to change existing fades, you have to delete them and reapply.  I also don't know if this will pass muster with the openshot devs, but it meets my current needs.  If interested, I can submit a pull request.

**Comment by Colorjet3** [comment]
Please connect with @jonoomph regarding this.  He is the lead developer and ultimately decision maker to approve/reject.
